test/Infer/pruning.opt

```diff
@@ -0,0 +1,25 @@
+; REQUIRES: solver, synthesis
+; RUN: %souper-check -infer-rhs -souper-exhaustive-synthesis -souper-dataflow-pruning -souper-exhaustive-synthesis-debug-level=2 %solver %s > %t1
+; RUN: %FileCheck %s < %t1
+
```

Comment:
I don't understand this test, how does it make sure that pruning occurred?

Reply:
The pruned RHS candidates get printed in this debug level. This test is looking for a few particular ones which are pruned.
Will a message like "Pruned" before each output make it better?

Reply:
I think so
right now the CHECK: lines just look like they're looking for regular RHS components, like any other test case we have, I think you should do something to make it clear these correspond to pruned branches of the search space (I would find this output very confusing if I were watching souper run)